### PR TITLE
boards: x86/xtensa: Add device tree support for USB 

### PR DESCRIPTION
**Pull request by SavinayDharmappa**

### CODE REVIEW

File: soc/xtensa/intel_s1000/soc.h [hunk 1]
```diff
@@ -183,7 +183,6 @@ struct soc_dsp_shim_regs {
 };
 
 #define USB_DW_BASE				0x000A0000
```
Comment (nashif) [inline soc/xtensa/intel_s1000/soc.h]: what about the base address, it is already define in DTS.

File: soc/x86/intel_quark/quark_se/soc.h [hunk 2]
```diff
@@ -77,7 +77,6 @@
 #ifdef CONFIG_USB_DW
 
 #define	USB_DW_BASE			QM_USB_0_BASE
```
Comment (nashif) [inline soc/x86/intel_quark/quark_se/soc.h]: shouldn't this come from DTS?

**Review by galak — CHANGES_REQUESTED** [review]
The compat should be either "snps,dwc2" or "snps,dwc3" depending on which version of the USB DW core is being used.

File: dts/x86/intel_curie.dtsi [hunk 3]
```diff
@@ -183,5 +183,14 @@
 			label = "WATCHDOG_0";
 
 		};
+
+		usbd: usbd@000a0000 {
```
Comment (nashif) [inline dts/x86/intel_curie.dtsi]: is this correct for this platform? QMSI has a different value.

```
ext/hal/qmsi/soc/quark_se/include/qm_soc_regs.h:#define QM_USB_0_BASE (0xB0500000)
```

**Comment by SavinayDharmappa** [comment]
recheck

**Comment by SavinayDharmappa** [comment]
@nashif  sanity is passing at my local machine not sure why failing github. I have cross checked tiwce

File: dts/x86/intel_curie.dtsi (hunk 3 above)
Comment (SavinayDharmappa) [reply dts/x86/intel_curie.dtsi]: done

**Comment by SavinayDharmappa** [comment]
resolved merge conflict

**Comment by SavinayDharmappa** [comment]
@galak could you please review it .

**Comment by ManojSubbarao** [comment]
@nashif please review

**Comment by ManojSubbarao** [comment]
@galak recheck

**Comment by ManojSubbarao** [comment]
@galak recheck

**Comment by rljordan** [comment]
@galak Can you please review this?  We would like to have Savinay respond to any further changes by end of week.  Thanks!

**Comment by carlescufi** [comment]
@galak @nashif is this still applicable?

**Comment by carlescufi** [comment]
@galak @nashif @finikorg is this still applicable?

**Comment by finikorg** [comment]
> @galak @nashif @finikorg is this still applicable?

Apparently support for all mentioned in the PR boards was removed, we can close this PR.

**Pull request closed without merging**